src/librustc/ty/layout.rs

```diff
@@ -1853,13 +1853,15 @@ impl<'a, 'tcx> TyLayout<'tcx> {
             ty::TyInt(_) |
             ty::TyUint(_) |
             ty::TyFloat(_) |
-            ty::TyFnPtr(_) |
-            ty::TyNever |
-            ty::TyFnDef(..) |
-            ty::TyDynamic(..) => {
+            ty::TyFnPtr(_) => {
                 bug!("TyLayout::field_type({:?}): not applicable", self)
```

Comment:
It mostly has to match the `Layout` variant (maybe I should match on that instead).